src/seedu/addressbook/commands/Command.java

```diff
@@ -70,4 +70,11 @@ public int getTargetIndex() {
     public void setTargetIndex(int targetIndex) {
         this.targetIndex = targetIndex;
     }
+
+    /**
+     * Indicates whether the command mutates data
+     */
+    public boolean isMutating() {
```

Comment:
Good attempt in not providing a default implementation since it is not immediately obvious whether a command should be mutating or not by default. Consider using an interface instead?